web/src/lib/components/shared-components/autogrow-textarea.svelte

```diff
@@ -0,0 +1,40 @@
+<script lang="ts">
+  import { autoGrowHeight } from '$lib/actions/autogrow';
+  import { clickOutside } from '$lib/actions/click-outside';
+  import { shortcut } from '$lib/actions/shortcut';
+  import { tick } from 'svelte';
+
+  export let content: string = '';
+  export let elementClass: string = '';
+  export let onContentUpdate: (newContent: string) => void = () => null;
+  export let placeholder: string = '';
+
+  let textarea: HTMLTextAreaElement;
+  $: newContent = content;
+
+  $: if (textarea) {
+    newContent;
+    void tick().then(() => autoGrowHeight(textarea));
+  }
+
+  const updateContent = () => {
+    if (content === newContent) {
+      return;
+    }
+    onContentUpdate(newContent);
+  };
+</script>
+
+<textarea
+  bind:this={textarea}
+  class={elementClass}
+  on:focusout={updateContent}
+  on:input={(e) => (newContent = e.currentTarget.value)}
+  use:clickOutside={{ onOutclick: updateContent }}
+  {placeholder}
+  use:shortcut={{
+    shortcut: { key: 'Enter', ctrl: true },
+    onShortcut: (e) => e.currentTarget.blur(),
+  }}
+  {...$$restProps}>{content}</textarea
```

Comment:
`data-testid` can be defined here and $$restProps wouldn't be needed

Reply:
My thought behind doing it this was two-fold:
* in case two instances of the component were rendered at the same time, it would be beneficial to pass the `data-testid` in so you can explicitly target one over another in a test
* futureproofing in case you wanted to be able to pass other props in (`disabled`, `readonly`, etc.) without having to modify the component's props

Happy to go with your suggestion if we don't feel like these are scenarios we'd have to account for.

Reply:
Gotcha. The downside of using `$$restProps` is the lack of type safety unless the `$$Props` type is defined. That's why I usually try to avoid using it. Maybe we can just declare the test id as a prop instead?

Reply:
Awesome, thank you!